test/suites/ubuntu-core-edgex-image/build-image.sh

```diff
@@ -0,0 +1,39 @@
+#!/bin/bash -ex
+
+if [[ -z "$KEY_NAME" ]]; then
+  echo "Required input 'KEY_NAME' is unset. Exiting..."
+  exit 1
+fi
+
+# Configure model assertion
+DEVELOPER_ID=$(snapcraft whoami | grep 'id:' | awk '{print $2}')
+TIMESTAMP=$(date -Iseconds --utc)
+yq e -i ".authority-id = \"$DEVELOPER_ID\"" model.yaml
+yq e -i ".brand-id = \"$DEVELOPER_ID\"" model.yaml
+yq e -i ".timestamp = \"$TIMESTAMP\"" model.yaml
+
+# Sign the model assertion
+yq eval model.yaml -o=json | snap sign -k $KEY_NAME > model.signed.yaml
+
+# Check the signed model
+cat model.signed.yaml
+
+# clean and build the image
+sudo rm -f pc.img
+ubuntu-image snap model.signed.yaml --validation=enforce --snap pc-gadget/pc_*_amd64.snap
+
+# check the image file
+file pc.img
+
+# test ubuntu core with QEMU
+sudo qemu-system-x86_64 \
+ -smp 4 \
+ -m 4096 \
```

Comment:
This is should not be a part of the image build.

This makes it hard to rerun the tests without rebuilding the image.

Reply:
I'd move this to run.sh and move everything related to running tests into a separate test.sh script.